rosetta-lang/src/main/java/com/regnosys/rosetta/formatting2/XtextResourceFormatter.java

```diff
@@ -48,16 +50,37 @@ public void formatCollection(Collection<Resource> resources, ITypedPreferenceVal
 
 	@Override
 	public void formatXtextResource(XtextResource resource, ITypedPreferenceValues preferenceValues) {
+		LOGGER.info("Formatting file at location " + resource.getURI());
+
 		// setup request and formatter
 		FormatterRequest req = formatterRequestProvider.get();
 		req.setPreferences(preferenceValues);
 		IFormatter2 formatter = iFormatter2Provider.get();
 
-		ITextRegionAccess regionAccess = regionBuilder.forNodeModel(resource).create();
+		ITextRegionAccess regionAccess = null;
+		try {
+			regionAccess = regionBuilder.forNodeModel(resource).create();
+		} catch (Exception e) {
+			LOGGER.info("Resource " + resource.getURI() + " is empty.");
+			return;
+		}
+
 		req.setTextRegionAccess(regionAccess);
 
 		// list contains all the replacements which should be applied to resource
-		List<ITextReplacement> replacements = formatter.format(req);
+		List<ITextReplacement> replacements = new ArrayList<>();
+		try {
+			replacements = formatter.format(req); // throws exception
+		} catch (RuntimeException e) {
+			LOGGER.error("RuntimeException in " + resource.getURI() + ": " + e.getMessage());
```

Comment:
When logging an error, always pass the error to the logger as well. I.e.,
```
LOGGER.error(msg, e);
```
=> the logger can then be configured whether or not to show the stack trace.